src/components/button/button.spec.tsx

```diff
@@ -0,0 +1,31 @@
+import Button from "./button"
+import { ButtonType } from '../constants/button-types'
+import { render, screen } from '@testing-library/react'
+import { faker } from '@faker-js/faker'
+
+describe('Button Component', () => {
+  test('Should render correctly', () => {
+    render(<Button variant={ButtonType.Success} label={faker.lorem.word()} />)
+    const button = screen.getByRole('button-' + ButtonType.Success)
+    expect(button).toBeInTheDocument()
+  })
+
+  test('Should call on click', () => {
+    const actionClicked = jest.fn()
+    render(<Button variant={ButtonType.Success} label={faker.lorem.word()} onClick={actionClicked} />)
+    const button = screen.getByRole('button-' + ButtonType.Success)
+    button.click()
+    expect(actionClicked).toHaveBeenCalledTimes(1)
+  })
+
+  test('Should display only one variant', () => {
+    render(<Button variant={ButtonType.Success} label={faker.lorem.word()} />)
+    const buttonSuccess = screen.queryByRole('button-' + ButtonType.Success)
+    expect(buttonSuccess).toBeInTheDocument()
+
+    const buttonAlert = screen.queryByRole('button-' + ButtonType.Alert)
+    expect(buttonAlert).not.toBeInTheDocument()
+    const buttonErro = screen.queryByRole('button-' + ButtonType.Erro)
```

Comment:
Não existe role customizável, a role é button. Depois dá uma olhada nesses dois artigos [About Queries](https://testing-library.com/docs/queries/about), e [W3C Recommendation: Accessible Name and Description](https://www.w3.org/TR/accname-1.1/)

Reply:
Uma dúvida, no caso do input, eu utilizo como role o "props.name" (da forma abaixo), levando em consideração que no meu teste eu também passo o nome do input como um fieldName (não sei se você reparou), isso teria problema?

`role={props.name}`

Reply:
Ajustes para utilizar a role="button" realizado em duas etapas:

commit: ec2e17718e44ed7c8cc7682a9919f1054d6d7ee2

Reply:
> Uma dúvida, no caso do input, eu utilizo como role o "props.name" (da forma abaixo), levando em consideração que > no meu teste eu também passo o nome do input como um fieldName (não sei se você reparou), isso teria problema?
> role={props.name}

Leia os artigos que eu linkei por favor, vai responder a sua duvida